service/matching/taskReader.go

```diff
@@ -207,8 +207,8 @@ Loop:
 
 		case <-updateAckTimer.C:
 			err := tr.persistAckLevel(ctx)
-			tr.tlMgr.signalIfFatal(err)
-			if err != nil {
+			isConditionFailed := tr.tlMgr.signalIfFatal(err)
+			if err != nil && !isConditionFailed {
```

Comment:
should it return early if `isconditionFailed`?

Reply:
In that case, `signalIfFatal` notifies the `taskQueueManager`, which cancels the context and makes this exit next time around the loop. I'd rather leave that decision in `taskQueueManager` instead of duplicating it here